### PR TITLE
Add ICP Index canister

### PR DESCRIPTION
**Pull request by lmuntaner**
Add ICP Index canister to snapshot.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@dskloetd please review

File: bin/dfx-nns-index-import [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+jq '.canisters["nns-index"] //= {
+    "build": "",
+    "candid": "candid/nns-index.did",
+    "type": "custom",
+    "wasm": "wasms/nns-index.wasm"
+  }' dfx.json | sponge dfx.json
+# ... Make sure that the internet_identity candid path is declared
+echo "Setting the candid path for the nns-index canister"
+NNS_INDEX_CANDID_PATH="$(jq -r '.canisters["nns-index"].candid' dfx.json)"
+echo "NNS_INDEX_CANDID_PATH=$NNS_INDEX_CANDID_PATH"
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply bin/dfx-nns-index-import]: Oops, nops.

File: bin/dfx-nns-index-import [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+test -n "${NNS_INDEX_CANDID_PATH:-}" || {
+  NNS_INDEX_CANDID_PATH="candid/nns-index.did"
+  mkdir -p "$(dirname "$NNS_INDEX_CANDID_PATH")"
+  NNS_INDEX_CANDID_PATH="$NNS_INDEX_CANDID_PATH" jq '.canisters["nns-index"].candid = (env.II_CANDID_PATH)' dfx.json | sponge dfx.json
+}
+
+echo "Downloading ICP Index canister .did file"
+# Download ICP Index canister .did file
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply bin/dfx-nns-index-import]: Removed.

File: bin/dfx-nns-index-install [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Install ICP index locally as documented in:
+# https://internetcomputer.org/docs/current/developer-docs/integrations/ledger/icp-index-local-setup
+
+LEDGER_ACCOUNT_ID=$(dfx canister id nns-ledger)
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply bin/dfx-nns-index-install]: Yes, I copied it and didn't realize it was called ACCOUNT. Changed.

File: bin/dfx-nns-index-install [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+dfx deploy nns-index --argument '(record {ledger_id = principal"'"${LEDGER_ACCOUNT_ID}"'";})'
+
+NNS_INDEX_CANISTER=$(dfx canister id nns-index)
+
+dfx canister call "$NNS_INDEX_CANISTER" ledger_id '()'
+
+dfx canister call "$NNS_INDEX_CANISTER" status '()'
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply bin/dfx-nns-index-install]: changed

File: bin/dfx-nns-index-install [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# Install ICP index locally as documented in:
+# https://internetcomputer.org/docs/current/developer-docs/integrations/ledger/icp-index-local-setup
+
+LEDGER_ACCOUNT_ID=$(dfx canister id nns-ledger)
+
+dfx deploy nns-index --argument '(record {ledger_id = principal"'"${LEDGER_ACCOUNT_ID}"'";})'
+
+NNS_INDEX_CANISTER=$(dfx canister id nns-index)
+
+dfx canister call "$NNS_INDEX_CANISTER" ledger_id '()'
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply bin/dfx-nns-index-install]: yes, done

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Review by dskloetd — APPROVED** [review]
Thanks!

**Pull request merged**